appeals/api/src/server/endpoints/documents/documents.middleware.js

```diff
@@ -1,20 +1,33 @@
 import { getDocumentById } from '#repositories/document.repository.js';
+import { upsertCaseFolders } from '#repositories/folder.repository.js';
 import { ERROR_NOT_FOUND } from '#endpoints/constants.js';
 
 /**
  * @type {import("express").RequestHandler}
  * @returns {Promise<object | void>}
  */
 export const validateDocumentAndAddToRequest = async (req, res, next) => {
-	const {
-		params: { guid }
-	} = req;
-	const document = await getDocumentById(guid);
+	const { documentId } = req.params;
+	const document = await getDocumentById(documentId);
 
 	if (!document) {
-		return res.status(404).send({ errors: { appealId: ERROR_NOT_FOUND } });
+		return res.status(404).send({ errors: { documentId: ERROR_NOT_FOUND } });
 	}
 
 	req.document = document;
 	next();
 };
+
+/**
+ * @type {import("express").RequestHandler}
+ * @returns {Promise<object | void>}
+ */
+export const ensureCaseFolders = async (req, res, next) => {
+	const { appeal } = req;
+	if (!appeal || !appeal.id || !appeal.reference) {
+		return res.status(404).send({ errors: { appealId: ERROR_NOT_FOUND } });
+	}
```

Comment:
The `checkAppealExistsAndAddToRequest` method will check if the appeal exists for the given `appealId` and return an error if not so it shouldn't be necessary to add an additional check here.

Reply:
Yeah, I'll remove though needed to make sure the appeal.reference is there. I suppose it will always be...

Reply:
Yes I think there should always be a reference.

However if we do need a specific check for the reference then perhaps we could add it to the generic function so it would then also check other requests.